cloudinit/distros/alpine.py

```diff
@@ -202,6 +205,344 @@ def preferred_ntp_clients(self):
+            "uid": "-u",
+        }
+
+        adduser_flags = {"system": "-S"}
+
+        # support kwargs having groups=[list] or groups="g1,g2"
+        groups = kwargs.get("groups")
+        if groups:
+            if isinstance(groups, str):
+                groups = groups.split(",")
+
+            if isinstance(groups, dict):
+                util.deprecate(
+                    deprecated=f"The user {name} has a 'groups' config value "
+                    "of type dict",
+                    deprecated_version="22.3",
+                    extra_message="Use a comma-delimited string or "
+                    "array instead: group1,group2.",
+                )
+
+            # remove any white spaces in group names, most likely
+            # that came in as a string like: groups: group1, group2
+            groups = [g.strip() for g in groups]
+
+            # kwargs.items loop below wants a comma delimeted string
+            # that can go right through to the command.
+            kwargs["groups"] = ",".join(groups)
+
+            primary_group = kwargs.get("primary_group")
+            if primary_group:
+                groups.append(primary_group)
+
+        if create_groups and groups:
+            for group in groups:
+                if not util.is_group(group):
+                    self.create_group(group)
+                    LOG.debug("created group '%s' for user '%s'", group, name)
+        if "uid" in kwargs:
+            kwargs["uid"] = str(kwargs["uid"])
+
+        unsupported_busybox_values: Dict[str, Any] = {
+            "groups": None,
+            "expiredate": None,
+            "inactive": None,
+            "passwd": None,
+        }
+
+        # Check the values and create the command
+        for key, val in sorted(kwargs.items()):
+            if key in adduser_opts and val and isinstance(val, str):
+                adduser_cmd.extend([adduser_opts[key], val])
+            elif (
+                key in unsupported_busybox_values
+                and val
+                and isinstance(val, str)
+            ):
+                # Busybox's 'adduser' does not support specifying these
+                # options so store them for use via alternative means.
+                if key == "groups":
+                    unsupported_busybox_values[key] = val.split(",")
+                else:
+                    unsupported_busybox_values[key] = val
+            elif key in adduser_flags and val:
+                adduser_cmd.append(adduser_flags[key])
+
+        # Don't create the home directory if directed so
+        # or if the user is a system user
+        if kwargs.get("no_create_home") or kwargs.get("system"):
+            adduser_cmd.append("-H")
+
+        # Busybox's 'adduser' puts username at end of command
+        adduser_cmd.append(name)
+
+        # Run the command
+        LOG.debug("Adding user %s", name)
+        try:
+            subp.subp(adduser_cmd)
+        except Exception as e:
```

Comment:
I think we expect this to be subp.ProcessExecutionError. We like to avoid generic Exception handling when we know what exceptions to deal with.
```suggestion
        except subp.ProcessExecutionError as e:
```

Reply:
Done.

This was lifted from add_user in __init__.py so, like some of @TheRealFalcon previous comments, the same improvements could be made there also